sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java

```diff
@@ -1366,14 +1368,36 @@ private CharSequence generateMainHeader(final String packageName)
                 "import %s;\n" +
                 "import %s;\n" +
                 "%s" +
+                "%s" +
                 "@javax.annotation.Generated(value = { \"uk.co.real_logic.sbe.generation.java.JavaGenerator\" })\n",
                 packageName,
                 fqMutableBuffer,
                 fqReadOnlyBuffer,
-                interfaceImportLine());
+                interfaceImportLine(),
+                generateClassJavaDoc(msgToken));
         }
     }
 
+    private static String generateClassJavaDoc(final Token msgToken)
+    {
+        final StringBuilder sb = new StringBuilder();
+        if (msgToken.description() != null)
+        {
+            sb.append(" * ").append(msgToken.description()).append("\n");
+        }
+
+        sb.append(" * ").append("@since ").append(msgToken.version()).append("\n");
```

Comment:
Version appears to always be `sinceVersion` even when `msgToken.versionContext` indicates otherwise.
Is this right or is there some circumstance I'm missing? This is on both `BEGIN_MESSAGE` and `BEGIN_FIELD` tokens.

Reply:
This is historical and no longer needed I believe.  Some of the early versions of SBE in the pre 1.0 phase may have used this but it is no longer.